src/connectionhealthpolicy/BaseConnectionHealthPolicy.ts

```diff
@@ -10,12 +10,18 @@ export default class BaseConnectionHealthPolicy implements ConnectionHealthPolic
   protected minHealth: number;
   protected maxHealth: number;
   protected currentHealth: number;
+  readonly name: string;
 
-  constructor(configuration: ConnectionHealthPolicyConfiguration, data: ConnectionHealthData) {
+  constructor(
+    configuration: ConnectionHealthPolicyConfiguration,
+    data: ConnectionHealthData,
+    name: string
```

Comment:
Since customer may be overriding these classes (we cannot say for sure), this would be a breaking change as we are introducing `name` as new required parameter. Could you make this `name` optional to preserve backward compatibility?

Reply:
Ah that makes sense, I initially planned to make it optional but didn't see a real need for it, but it makes sense that customers may implement the interface or extend this class. Will change!